### ISSUE TITLE
Not able to provide alt text for images in Vega Mark

### ISSUE THREAD

**Issue by sambaran69** [vega/vega#1352]
When creating a Vega Mark of the Type Image, I can specify a URL but there is no option to specify the ALT text if vega is unable to download the image.

**Comment by jheer** [comment]
There is no standard for providing alt text on an SVG image nor for a sub-region of a Canvas. However, there are two other strategies you might try:

1. Use the `tooltip` channel to provide useful info that can also serve as alt text. This will work for all renderers. In the case of SVG, a `title` tag will be included within the SVG `text`.
2. Add an additional Vega text mark to create your own alt text that is directly visible (not just via tooltip) on the display, using reactive geometry (that is, the text marks use the image marks as their backing data). If an image fails to load, the scenegraph item will have a `item.image.loaded` flag set to `false`. You can use this to drive a text encoding; the only catch is that image loading is asynchronous and the encoding channels other than `image` are not automatically re-evaluated once the image is loaded (or fails to load). As a result, in the example below you need to hover over the missing image to trigger a re-evaluation, at which point the "alt text" is shown.

Here's an example of both 1 & 2 above, based on a modified version of the `image.vg.json` test specification:

```json
{
  "$schema": "https://vega.github.io/schema/vega/v4.json",
  "width": 200,
  "height": 200,
  "padding": 5,

  "data": [
    {
      "name": "data",
      "values": [
        {"x": 0.5, "y": 0.5, "img": "data/ffox.png"},
        {"x": 1.5, "y": 1.5, "img": "data/gimp.png"},
        {"x": 2.5, "y": 2.5, "img": "data/7zip_blah.png"}
      ]
    }
  ],

  "scales": [
    {
      "name": "x",
      "type": "linear",
      "domain": [0, 3],
      "range": "width"
    },
    {
      "name": "y",
      "type": "linear",
      "domain": [0, 3],
      "range": "height"
    }
  ],

  "axes": [
    {"orient": "bottom", "scale": "x", "tickCount": 5},
    {"orient": "left", "scale": "y", "tickCount": 5}
  ],

  "marks": [
    {
      "type": "image",
      "name": "images",
      "from": {"data": "data"},
      "encode": {
        "enter": {
          "url": {"field": "img"},
          "tooltip": {"field": "img"},
          "width": {"value": 50},
          "height": {"value": 50},
          "x": {"scale": "x", "field": "x"},
          "y": {"scale": "y", "field": "y"},
          "align": {"value": "center"},
          "baseline": {"value": "middle"}
        },
        "update": {
          "opacity": {"value": 1.0}
        },
        "hover": {
          "opacity": {"value": 0.5}
        }
      }
    },
    {
      "type": "text",
      "from": {"data": "images"},
      "encode": {
        "enter": {
          "x": {"field": "x"},
          "y": {"field": "y"},
          "align": {"value": "center"}
        },
        "update": {
          "text": {"signal": "datum.image && datum.image.loaded === false ? datum.datum.img : ''"}
        }
      }
    }
  ]
}
```
